kill-developer.py

```diff
@@ -1,57 +1,98 @@
 #!/usr/bin/env python
 # vim:se fileencoding=utf8 noet :
 # (c) 2017 Michał Górny
+# (c) 2018 Amy Liffey
 # 2-clause BSD license
 
+import argparse
 from collections import defaultdict, namedtuple
 import errno
 import glob
 import io
 import json
 from lxml.builder import E
-import lxml.etree
+import lxml.etree as etree
 import os
 import os.path
 import sys
 
 
-def main(repopath, dev_email):
-	if '@' not in dev_email:
-		dev_email += '@gentoo.org'
+def main() -> int:
+    """Parameters"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-p', '--path', help='Path to your repo /home/user/gentoo/')
+    parser.add_argument('-e', '--email', help='Email of person you want to retire user@gentoo.org')
+    parser.add_argument('-r', '--repoman', help='Add if you want to run repoman', action='store_true')
+    args = parser.parse_args()
 
-	for f in glob.glob(os.path.join(repopath, '*/*/metadata.xml')):
-		subpath = os.path.relpath(f, repopath)
+    if args.email is None or args.path is None:
+        parser.print_help(sys.stdout)
+        return 0
 
-		xml = lxml.etree.parse(f)
-		r = xml.getroot()
+    # Store packages which are maintained by the person
+    proxy_maint = False
+    if '@gentoo.org' not in args.email:
+        proxy_maint = True
+    pkg = []
+    grabs = []
 
-		# check if dev maintains it
-		maints = r.findall('maintainer')
-		for m in maints:
-			if m.findtext('email') == dev_email:
-				break
-		else:
-			continue
+    for f in glob.glob(os.path.join(args.path, '*/*/metadata.xml')):
+        # Store subpath, parse xml
+        subpath = os.path.relpath(f, args.path)
+        xml = etree.parse(f)
+        r = xml.getroot()
 
-		# a single maintainer? we need maintainer-needed now!
-		if len(maints) == 1:
-			c = lxml.etree.Comment(' maintainer-needed ')
-			c.tail = m.tail
-			r.replace(m, c)
+        # Check if person maintains the package
+        maints = r.findall('maintainer')
 
-			# print for 'up for grabs'
-			print('/'.join(subpath.split('/')[:2]))
-		else:
-			r.remove(m)
+        for m in maints:
+            if m.findtext('email') == args.email:
+                pkg.append('/'.join(subpath.split('/')[:2]))
+                break
+        else:
+            continue
 
-		with open(f, 'wb') as f:
-			f.write(b'<?xml version="1.0" encoding="UTF-8"?>\n')
-			xml.write(f, encoding='UTF-8')
-			# yay, add trailing newline because lxml is dumb
-			f.write(b'\n')
+        # a single maintainer? we need maintainer-needed now!
+        if proxy_maint and len(maints) == 2:
```

Comment:
Since this is script for proxy maint project we do not expect that we retire people who do not have proxy-maint project in metadata.. but valid point .. I am afraid its not necessary